samples/net/mqtt/src/modules/led/led.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+#include <zephyr/drivers/led.h>
+
+#include "message_channel.h"
+
+/* Register log module */
+LOG_MODULE_REGISTER(led, CONFIG_MQTT_SAMPLE_LED_LOG_LEVEL);
+
+const static struct device *led_device = DEVICE_DT_GET_ANY(gpio_leds);
+
+/* LED 1, green on Thingy:91 boards. */
+#define LED_1_GREEN 1
+
+void led_callback(const struct zbus_channel *chan)
+{
+	const enum network_status *status;
+
+	if (&NETWORK_CHAN == chan) {
+
+		/* Get network status from channel. */
+		status = zbus_chan_const_msg(chan);
+
+		switch (*status) {
+		case NETWORK_CONNECTED:
+			__ASSERT_NO_MSG(led_on(led_device, LED_1_GREEN) == 0);
```

Comment:
I don't think we can have asserts here. LED won't work if asserts are disabled (which is typically the default case)

Reply:
Your right, if assert are disabled the function will never be called.